### PR TITLE
Remove old deprecations from 1.30 and earlier

### PR DESCRIPTION
**Pull request by lydia-duncan**
Removes deprecated versions of:
- fileReader.flush
- openfp
- openfd
- bigint initializers that halt or return error codes
- file.check
- opentmp
- fileReader.seek and fileWriter.seek
- openreader/openReader
- file.reader
- file.lines
- file.writer

Deprecates flags that had been used to control the changed behavior:
- bigintInitThrows
- useNewSeekRegionBounds
- useNewOpenReaderRegionBounds
- useNewFileReaderRegionBounds
- useNewLinesRegionBounds
- useNewFileWriterRegionBounds

Note: I think I still have deprecations to remove, but will split them into a
separate PR in the interest of getting some removals in this release, since
there's already a lot here.

A couple of the changes meant that some helper functions had extra
arguments that were no longer needed - removed those arguments, which
meant updating the URL module in addition to the files with the
deprecations.

Updates tests that were setting the now-deprecated flags to stop setting
them.  In many cases, this meant that the .compopts/COMPOPTS files
could be removed entirely, so removed them.

Some of the submitted shootouts ended up failing as a result of the openfd
removal.  Since these tests should not be modified until new versions have
been submitted, add a .notest for them until they can be updated.

Removes tests of the deprecated behavior.  For the newly deprecated flags,
adds tests to lock in the deprecation messages

Passed a full paratest with futures.  Passed a performance test over the
SSCA2 changes.

### CODE REVIEW

**Comment by lydia-duncan** [comment]
Note for reviewer: this is probably easiest to review on a commit-by-commit basis

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -5422,7 +5224,8 @@ proc file.reader(param kind=iokind.dynamic, param locking=true,
    the :type:`fileReader`.  When set to ``false``, the region argument will
    exclude the high bound.  Defaults to ``false``, the original behavior.
  */
```
Comment (ShreyasKhandekar) [inline modules/standard/IO.chpl:5226]: Remove "when true" and "when false" parts of the documentation here as well

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -5580,7 +5339,8 @@ proc file.lines(param locking:bool = true, start:int(64) = 0,
    this function would cover.  When set to ``false``, the region argument will
    exclude the high bound.  Defaults to ``false``, the original behavior.
  */
```
Comment (ShreyasKhandekar) [inline modules/standard/IO.chpl:5341]: Remove "when true" and "when false" here as well

File: modules/standard/IO.chpl [hunk 3]
```diff
@@ -5668,7 +5407,8 @@ proc file.writer(param kind=iokind.dynamic, param locking=true,
    exclude the high bound.  Defaults to ``false``, the original behavior.
 
  */
```
Comment (ShreyasKhandekar) [inline modules/standard/IO.chpl:5409]: Another instance of the "when true/false" stuff, please remove them here too.

**Review by ShreyasKhandekar — APPROVED** [review]
Looks good! Just some minor consistency suggestions

**Review by ShreyasKhandekar — APPROVED** [review]

**Comment by ShreyasKhandekar** [comment]
I have a follow up question:
Were `openWriter` with a style argument and `openwriter` deprecated in different cycles compared to `openReader` with a style argument and `openreader` ?
I saw you only remove the latter even though both were deprecated. 

Please ignore if you're gonna follow it up in a subsequent PR :)

**Comment by lydia-duncan** [comment]
Style arguments have had a mixed history - they were marked deprecated, then unstable, then deprecated again in between when this feature was changed.  And openwriter doesn't have a region argument.  So I thought it was appropriate to remove the style argument versions that were tied to this particular adjustment.  There should still be style versions for all the IO functions, just slightly above the other versions iirc.

**Comment by lydia-duncan** [comment]
Basically, iirc the latest round of deprecation for style arguments was too recent to warrant removing them.

**Pull request merged**